diffcp/cone_program.py

```diff
@@ -131,31 +139,39 @@ def solve_and_derivative(A, b, c, cone_dict, warm_start=None, **kwargs):
     m, n = A.shape
     N = m + n + 1
     cones = cone_lib.parse_cone_dict(cone_dict)
+    cones_parsed = cone_lib.parse_cone_dict_cpp(cones)
     z = (x, y - s, np.array([1]))
     u, v, w = z
+
     Q = sparse.bmat([
         [None, A.T, np.expand_dims(c, - 1)],
         [-A, None, np.expand_dims(b, -1)],
         [-np.expand_dims(c, -1).T, -np.expand_dims(b, -1).T, None]
     ])
-    D_proj_dual_cone = _diffcp.dprojection(
-        v, cone_lib.parse_cone_dict_cpp(cones), True)
+    Q_dense = Q.todense()
```

Comment:
`Q_dense` should only be formed when `mode == "dense"`

Reply:
Oops, moved